worlds/dkc2/Rom.py

```diff
@@ -0,0 +1,715 @@
+            line = line.replace("LEVEL", hint.level)
+            line = line.replace("WORLD", hint.world)
+            line = line.replace("NUM", hint.bonus)
+            line = line.replace("PLAYER", hint.player)
+            line = line.replace("ITEM", hint.item)
+            line = line.center(32, " ").rstrip() + "°"
+            line = string_to_bytes(line)
+            hint_data += line
+            addr += len(line)
+
+        count += 1
+        if count == 41:
+            break
+
+    patch.write_file("cranky_hints.bin", hint_data)
+    patch.write_file("cranky_hint_offsets.bin", hint_offsets)
+
+
+def compute_wrinkly_hints(world: "DKC2World", patch: DKC2ProcedurePatch):
+    hint_data = bytearray()
+    hint_offsets = bytearray()
+    hints = []
+
+    hintable_items = [
+        ItemName.diddy,
+        ItemName.dixie,
+        ItemName.crocodile_cauldron,
+        ItemName.krem_quay,
+        ItemName.krazy_kremland,
+        ItemName.gloomy_gulch,
+        ItemName.krools_keep,
+        ItemName.the_flying_krock,
+        ItemName.lost_world_cauldron,
+        ItemName.lost_world_quay,
+        ItemName.lost_world_kremland,
+        ItemName.lost_world_gulch,
+        ItemName.lost_world_keep,
+        ItemName.carry,
+        ItemName.climb,
+        ItemName.cling,
+        ItemName.cartwheel,
+        ItemName.swim,
+        ItemName.team_attack,
+        ItemName.helicopter_spin,
+        ItemName.rambi,
+        ItemName.squawks,
+        ItemName.enguarde,
+        ItemName.squitter,
+        ItemName.rattly,
+        ItemName.clapper,
+        ItemName.glimmer,
+        ItemName.barrel_kannons,
+        ItemName.barrel_exclamation,
+        ItemName.barrel_kong,
+        ItemName.barrel_warp,
+        ItemName.barrel_control,
+        ItemName.skull_kart,
+        ItemName.lost_world_rock,
+        ItemName.dk_coin,
+        ItemName.kremkoins,
+        ItemName.banana_coin,
+        ItemName.red_balloon,
+    ]
+    unskippable_items = [
+        ItemName.lost_world_rock,
+        ItemName.dk_coin,
+        ItemName.kremkoins,
+        ItemName.banana_coin,
+        ItemName.red_balloon,
+    ]
+
+    abort = False
+    count = 0
+    for item in hintable_items:
+        if item in world.options.start_inventory.value and item not in unskippable_items:
+            continue
+        locations = world.multiworld.find_item_locations(item, world.player)
+        if len(locations) == 0:
+            continue
+        
+        for location in locations:
+            hint = WrinklyHint(
+                location.item.name,
+                location.name,
+                world.multiworld.get_player_name(location.player),
+                location.game,
+            )
+            hints.append(hint)
+            count += 1
+            if count == 28:
+                abort = True
+                break
+
+        if abort:
+            break
+
+    world.random.shuffle(hints)
+    # B4AC84
+    addr = 0
+    hint: WrinklyHint
```

Comment:
```suggestion
```
This type hint isn't necessary since that's already the type from the prior use of `hint`, and this way gives an error for reassigning.